src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2835,6 +2835,24 @@ LogicalResult ONNXSplitV11Op::inferShapes(
 // Flatten
 //===----------------------------------------------------------------------===//
 
+static LogicalResult verify(ONNXFlattenOp op) {
+
+  auto inTy = op.input().getType().dyn_cast<RankedTensorType>();
```

Comment:
That will not work when we use memref. In verify, we have to accommodate them as well. That is why we test with 
```
  if (!hasShapeAndRank(op.input())) {
```

then, if you need the type, you should use
```
  auto inTy = op.input().getType().dyn_cast<ShapedTensorType>();
```